provider/marathon/marathon.go

```diff
@@ -537,3 +676,12 @@ func parseIndex(index string, length int) (int, error) {
 
 	return parsed, nil
 }
+
+func extractServiceName(varg ...string) string {
+	switch {
+	case len(varg) == 1:
+		return varg[0]
+	default:
```

Comment:
I wouldn't mind preserving the method signatures if it came practically free of charge. In this case though, I believe the workaround to achieve compatibility isn't very light-weight, so I'd rather go with a breaking change.

Our template guarantees might change in the future; for now though, I think we should take advantage of the flexibility we have if it helps us.